### PR TITLE
refactor(alert): add code type

### PR DESCRIPTION
**Pull request by liaoliaojun**

### CODE REVIEW

File: packages/alert/src/index.vue [hunk 1]
```diff
@@ -71,16 +71,16 @@ export default defineComponent({
...
     // state
-    const visible = ref(true)
+    const visible = ref<boolean>(true)
 
     // computed
-    const typeClass = computed(() => `el-alert--${ props.type }`)
-    const iconClass = computed(() => TYPE_CLASSES_MAP[props.type] || 'el-icon-info')
-    const isBigIcon = computed(() => props.description || ctx.slots.default ? 'is-big' : '')
-    const isBoldTitle = computed(() => props.description || ctx.slots.default ? 'is-bold' : '')
+    const typeClass = computed<string>(() => `el-alert--${ props.type }`)
```
Comment (zouhangwithsweet) [inline packages/alert/src/index.vue:77]: I think it's useless code. TypeScript can deduce the correct type.

File: packages/alert/src/index.vue [hunk 2]
```diff
@@ -71,16 +71,16 @@ export default defineComponent({
...
+    const iconClass = computed<string>(() => TYPE_CLASSES_MAP[props.type] || 'el-icon-info')
+    const isBigIcon = computed<string>(() => props.description || ctx.slots.default ? 'is-big' : '')
+    const isBoldTitle = computed<string>(() => props.description || ctx.slots.default ? 'is-bold' : '')
 
     // methods
-    const close = evt => {
+    const close:(evt: HTMLElement) => void = evt => {
```
Comment (zouhangwithsweet) [inline packages/alert/src/index.vue:83]: why not just declare type for `evt`

**Review by zouhangwithsweet — CHANGES_REQUESTED** [review]

File: packages/alert/src/index.vue (hunk 1 above)
Comment (JeremyWuuuuu) [reply packages/alert/src/index.vue:77]: true

File: packages/alert/src/index.vue [hunk 3]
```diff
@@ -71,16 +71,16 @@ export default defineComponent({
   emits: ['click'],
   setup(props, ctx) {
     // state
-    const visible = ref(true)
+    const visible = ref<boolean>(true)
```
Comment (JeremyWuuuuu) [inline packages/alert/src/index.vue:74]: Same here.

File: packages/alert/src/index.vue [hunk 4]
```diff
@@ -71,16 +71,16 @@ export default defineComponent({
...
 
     // computed
-    const typeClass = computed(() => `el-alert--${ props.type }`)
-    const iconClass = computed(() => TYPE_CLASSES_MAP[props.type] || 'el-icon-info')
-    const isBigIcon = computed(() => props.description || ctx.slots.default ? 'is-big' : '')
-    const isBoldTitle = computed(() => props.description || ctx.slots.default ? 'is-bold' : '')
+    const typeClass = computed<string>(() => `el-alert--${ props.type }`)
+    const iconClass = computed<string>(() => TYPE_CLASSES_MAP[props.type] || 'el-icon-info')
+    const isBigIcon = computed<string>(() => props.description || ctx.slots.default ? 'is-big' : '')
+    const isBoldTitle = computed<string>(() => props.description || ctx.slots.default ? 'is-bold' : '')
```
Comment (JeremyWuuuuu) [inline packages/alert/src/index.vue:80]: Same, I wonder why do you have to add types here?

**Comment by JeremyWuuuuu** [comment]
The PR does not making any sense to me, seems that you need to check out typescript documentations before get your hands on this. Please do fill me in if you need any help

**Pull request closed without merging**